posthog/tasks/test/test_stop_surveys_reached_target.py

```diff
@@ -0,0 +1,130 @@
+from posthog.models import Survey, Organization, Team, User, FeatureFlag
+from django.test import TestCase
+from dateutil.relativedelta import relativedelta
+from datetime import timedelta
+from django.utils.timezone import now
+from posthog.test.base import _create_event, flush_persons_and_events, ClickhouseTestMixin
+from posthog.tasks.stop_surveys_reached_target import stop_surveys_reached_target
+
+
+class TestStopSurveysReachedTarget(TestCase, ClickhouseTestMixin):
+    def setUp(self) -> None:
+        super().setUp()
+
+        self.org = Organization.objects.create(name="Org 1")
+        self.team1 = Team.objects.create(organization=self.org, name="Team 1")
+        self.team2 = Team.objects.create(organization=self.org, name="Team 2")
+        self.user = User.objects.create_and_join(self.org, "a@b.c", password=None)
+        self.flag = FeatureFlag.objects.create(
+            team=self.team1,
+            created_by=self.user,
+            key="flag_name",
+            filters={},
+            rollout_percentage=100,
+        )
+
+    def _create_event_for_survey(self, survey: Survey, event: str = "survey sent") -> None:
+        _create_event(
+            distinct_id="0",
+            event=event,
+            properties={
+                "$survey_id": str(survey.id),
+            },
+            timestamp=now(),
+            team=survey.team,
+        )
+
+    def test_stop_surveys_with_enough_responses(self) -> None:
+        surveys = [
+            Survey.objects.create(
+                name="1",
+                team=self.team1,
+                created_by=self.user,
+                linked_flag=self.flag,
+                responses_limit=1,
+                created_at=now(),
+            ),
+            Survey.objects.create(
+                name="2",
+                team=self.team1,
+                created_by=self.user,
+                linked_flag=self.flag,
+                responses_limit=1,
+                created_at=now(),
```

Comment:
hmm, would like to see a test for the earliest start date logic working accurately too. One way to test this would be to add a valid event that happens before the start date and ensuring its not counted in responses.

This reminds me of another annoying problem, the earliest start date won't be accurate if we restart the survey, so we need to add a 'first start date' to the survey. Maybe not as a part of this PR, since this is getting big already, but just add a TODO where the logic is for getting earliest start date?

Reply:
> hmm, would like to see a test for the earliest start date logic working accurately too. One way to test this would be to add a valid event that happens before the start date and ensuring its not counted in responses.

Done

> This reminds me of another annoying problem, the earliest start date won't be accurate if we restart the survey, so we need to add a 'first start date' to the survey. Maybe not as a part of this PR, since this is getting big already, but just add a TODO where the logic is for getting earliest start date?


Argh, my naming is incorrect, I'm not using the earliest start date, I'm using the earliest creation date (exactly for the reason from your second paragraph). Does this make sense now? I renamed earliest_survey_start_date -> earliest_survey_creation_date

Reply:
ooh excellent, that's much nicer 🙌